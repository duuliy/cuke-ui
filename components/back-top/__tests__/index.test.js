import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import BackTop from "../index";

describe("<BackTop/>", () => {
  it("should render a <BackTop/> components", () => {
    const wrapper = render(
      <div>
        <BackTop />
        <BackTop>
          <div>backTop</div>
        </BackTop>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find cuke-back-top classnames", () => {
    const wrapper = render(
      <div>
        <BackTop />
        <BackTop>
          <div>backTop</div>
        </BackTop>
      </div>
    );
    assert(wrapper.find(".cuke-back-top").length >= 1);
    assert(wrapper.find(".cuke-back-top-inner").length >= 1);
    assert(wrapper.find(".cuke-back-top-inner-icon").length === 1);
  });

  it("should render custom text", () => {
    const wrapper = render(
      <div>
        <BackTop />
        <BackTop>
          <div>哈哈哈</div>
        </BackTop>
      </div>
    );
    expect(wrapper.text()).toEqual("哈哈哈");
  });
  it("should can trigger onClick event", () => {
    const onClick = jest.fn();
    const wrapper = shallow(<BackTop onClick={onClick} />);
    wrapper.find(".cuke-back-top-inner").simulate("click");
    expect(onClick).toHaveBeenCalled();
  });

  it("should show back to element when scroll", () => {
    const wrapper = shallow(<BackTop visibilityHeight={100} />);
    assert(wrapper.find(".cuke-back-top-close").length === 1);

    window.scrollTo(0, 200);
    setTimeout(() => {
      assert(wrapper.find(".cuke-back-top-open").length === 1);
    });
  });

  it("should window cannot trigger click event when un mount", () => {
    const onClick = jest.fn();
    const wrapper = shallow(<BackTop />);
    wrapper.unmount();
    window.onclick = () => onClick;
    expect(onClick).not.toHaveBeenCalled();
  });

  it("should emit bind scroll", () => {
    const wrapper = new BackTop({ visibilityHeight: 100 }, { visible: null });
    wrapper.bindScroll();
    assert(true);
  });
});
