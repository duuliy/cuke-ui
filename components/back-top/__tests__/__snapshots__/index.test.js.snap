// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackTop/> should render a <BackTop/> components 1`] = `
<div>
  <div
    class="cuke-back-top cuke-back-top-close cuke-ui-no-animate"
  >
    <div
      class="cuke-back-top-inner"
    >
      <div
        class="cuke-back-top-inner-icon"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <polyline
            points="18 15 12 9 6 15"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="cuke-back-top cuke-back-top-close cuke-ui-no-animate"
  >
    <div
      class="cuke-back-top-inner"
    >
      <div>
        backTop
      </div>
    </div>
  </div>
</div>
`;
